File: lobwebapp-core/src/main/java/com/espindola/lobwebapp/service/impl/DefaultProductService.java
package com.espindola.lobwebapp.service.impl;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.espindola.lobwebapp.domain.Product;
import com.espindola.lobwebapp.repository.contract.ProductRepository;
import com.espindola.lobwebapp.service.contract.ProductService;
import com.espindola.lobwebapp.service.impl.base.AbstractEntityService;

@Service
public class DefaultProductService extends AbstractEntityService<Product>
		implements ProductService {

	private ProductRepository repository;

	@Autowired
	public DefaultProductService(ProductRepository repository) {
		super(repository);
		this.repository = repository;
	}

	@Override
	public List<Product> find(String name) {
		return repository.findByName(name);
	}

}

File: lobwebapp-html/src/main/ts/lwa/service/impl/DefaultProductService.ts
///<reference path='../../../DefinitelyTyped/angularjs/angular.d.ts'/>
///<reference path='../../domain/Product.ts'/>
///<reference path='../contract/ProductService.ts'/>
///<reference path='base/AbstractEntityService.ts'/>

module lwa.service.impl{
    import domain = lwa.domain;
    import service = lwa.service;
    
	export class DefaultProductService extends service.impl.base.AbstractEntityService<domain.Product> implements service.contract.ProductService {
        
        constructor($http : ng.IHttpService){
            super('product', $http);
        }
        
        public findByName (name : string,
            successCallback: (data: domain.Product[], status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any, 
            errorCallback: (data: domain.Product[], status: number, headers: (headerName: string) => string, config: ng.IRequestConfig)=> any) 
            {
            }                                   
       
        
    }
}